datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -192,10 +192,17 @@ def default_json_event_encoding(o):
     raise TypeError
 
 
-def obfuscate_sql_with_metadata(query, options=None):
+def obfuscate_sql_with_metadata(query, options=None, replace_null_character=False):
     if not query:
         return {'query': '', 'metadata': {}}
 
+    if replace_null_character:
+        # replace embedded null characters \x00 before obfuscating
+        # WHY NOT ALWAYS REPLACE?
```

Comment:
We've seen Postgres integration obfuscates up to 10k (based on the default config we documented) queries. Depends on the cardinality of the monitored database and the size of the query, the amount of query text to obfuscate could be huge. Obfuscation is CPU intensive and holds GIL, I think it's better to avoid operations that are not needed.

Reply:
Hmm, won't we **have to** drop nulls in the case of Postgres obfuscation because Postgres doesn't accept null characters? Maybe I didn't understand the second part of your comment.